ui.apps/src/main/content/jcr_root/apps/authoring-toolkit/depends-on/js/actions/dependsOnUpdateOptionsAction.js

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Action to refresh the option set of a Coral3 Select widget with an OptionProvider.
+ * Used in cases when path to options node is authored via another widget. Change of this widget's value triggers s
+ * new query for options
+ */
+(function (Granite, $, DependsOn) {
+
+    'use strict';
+
+    /**
+     * Finds out whether the two option sets are equal, e.g. to check if the
+     * newly collected options are the same as options already present in the option container
+     *
+     * @param optionSet0 First option set
+     * @param optionSet1 Second option set
+     * @returns {boolean}
+     */
+    function optionsAreSame(optionSet0, optionSet1) {
```

Comment:
`DependsOn.isEqual` can deeply compare two arrays

Reply:
Didn't know that) fixed